kpops/components/streams_bootstrap_v3/streams/model.py

```diff
@@ -0,0 +1,284 @@
+    )
+    idle_replicas: int | None = Field(
+        default=None,
+        title="Idle replica count",
+        description=describe_attr("idle_replicas", __doc__),
+    )
+    internal_topics: list[str] = Field(
+        default=[],
+        description=describe_attr("internal_topics", __doc__),
+    )
+    topics: list[str] = Field(
+        default=[],
+        description=describe_attr("topics", __doc__),
+    )
+    additional_triggers: list[str] = Field(
+        default=[],
+        description=describe_attr("additional_triggers", __doc__),
+    )
+    model_config = ConfigDict(extra="allow")
+
+
+class PersistenceConfig(BaseModel):
+    """streams-bootstrap persistence configurations.
+
+    :param enabled: Whether to use a persistent volume to store the state of the streams app.
+    :param size: The size of the PersistentVolume to allocate to each streams pod in the StatefulSet.
+    :param storage_class: Storage class to use for the persistent volume.
+    """
+
+    enabled: bool = Field(
+        default=False,
+        description="Whether to use a persistent volume to store the state of the streams app.	",
+    )
+    size: str | None = Field(
+        default=None,
+        description="The size of the PersistentVolume to allocate to each streams pod in the StatefulSet.",
+    )
+    storage_class: str | None = Field(
+        default=None,
+        description="Storage class to use for the persistent volume.",
+    )
+
+    @model_validator(mode="after")
+    def validate_mandatory_fields_are_set(
+        self: PersistenceConfig,
+    ) -> PersistenceConfig:  # TODO: typing.Self for Python 3.11+
+        if self.enabled and self.size is None:
+            msg = (
+                "If app.persistence.enabled is set to true, "
+                "the field app.persistence.size needs to be set."
+            )
+            raise ValidationError(msg)
+        return self
+
+
+class StreamsAppValues(StreamsBootstrapV3Values):
+    """streams-bootstrap app configurations.
+
+    The attributes correspond to keys and values that are used as values for the streams bootstrap helm chart.
+
+    :param kafka: streams-bootstrap kafka section
+    :param autoscaling: Kubernetes event-driven autoscaling config, defaults to None
+    """
+
+    kafka: StreamsConfig = Field(
+        default=...,
+        description=describe_attr("kafka", __doc__),
+    )
+    autoscaling: StreamsAppAutoScaling | None = Field(
+        default=None,
+        description=describe_attr("autoscaling", __doc__),
+    )
+    stateful_set: bool = Field(
+        default=False,
+        description="Whether to use a Statefulset instead of a Deployment to deploy the streams app.",
+    )
+    persistence: PersistenceConfig = Field(
+        default=PersistenceConfig(),
+        description=describe_attr("persistence", __doc__),
+    )
+    model_config = ConfigDict(extra="allow")
+
+    @model_validator(mode="after")
+    def validate_mandatory_fields_are_set(
```

Comment:
I appreciate the effort but shouldn't we delegate this validation to the helm chart?

Reply:
As far as I remember, this check wasn't included in the Helm charts for [v2](https://github.com/bakdata/streams-bootstrap/blob/1a0f2a490e6d613fa4356cd42541009273bd511d/charts/streams-app/templates/scaled-object.yaml#L30).

Neither for [v3](https://github.com/bakdata/streams-bootstrap/blob/935b61d79f5022420a5add51975a9ffecb222292/charts/streams-app/templates/scaled-object.yaml#L30)

Reply:
You are right but I think it is better to submit a PR there :)

Reply:
https://github.com/bakdata/streams-bootstrap/pull/254

Reply:
@philipp94831 Although you can argue that this makes it fail earlier in the generate step rather than the dry-run deployment step